### PR TITLE
[ACR] New command groups cache and credential-set

### PR DESCRIPTION
**Pull request by mabenedi**
- New command group `az acr cache` to manage cache rules
- New command group `az acr credential-set` to manage credential sets.
- Bump SDK version to 10.1.0

**Testing Guide**
View help file and tests.

**History Notes**
[ACR] `az acr cache`: Add new command group to manage cache rules
[ACR] `az acr credential-set`: Add new command group to manage credential sets

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
acr

**Comment by mabenedi** [comment]
Hey team, trying to merge this one before code complete next week.

File: src/azure-cli/azure/cli/command_modules/acr/cache.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                     client,
+                     registry_name,
+                     name,
+                     source_repo,
+                     target_repo,
+                     cred_set=None):
+
+    registry, rg = get_registry_by_name(cmd.cli_ctx, registry_name, None)
+
+    cred_set_id = AzureCoreNull if not cred_set else f'{registry.id}/credentialSets/{cred_set}'
```
Comment (rosanch) [inline src/azure-cli/azure/cli/command_modules/acr/cache.py:55]: what is the f'' for?

**Review by rosanch — CHANGES_REQUESTED** [review]
LGTM. But I'd request a PM to review the wording on the help messages, command arguments descriptions and error messages before merging.

File: src/azure-cli/azure/cli/command_modules/acr/cache.py (hunk 1 above)
Comment (mabenedi) [reply src/azure-cli/azure/cli/command_modules/acr/cache.py:55]: It's just string interpolation (python calls them f-strings) to create the credential set resource id

File: src/azure-cli/azure/cli/command_modules/acr/_help.py [hunk 2]
```diff
@@ -549,6 +549,100 @@
...
+type: command
+short-summary: Create a credential set.
+examples:
+  - name: Create a credential set.
+    text: az acr credential-set create -r MyRegistry -n MyRule -l docker.io -u https://MyKeyvault.vault.azure.net/secrets/usernamesecret -p https://MyKeyvault.vault.azure.net/secrets/passwordsecret
+"""
+
+helps['acr credential-set update'] = """
+type: command
+short-summary: Update the username or password KV secret ID on a credential set.
```
Comment (luisdlp) [inline src/azure-cli/azure/cli/command_modules/acr/_help.py:632]: nit: do we want to use "Azure Key Vault" instead of the "KV" acronym?

Comment (mabenedi) [reply src/azure-cli/azure/cli/command_modules/acr/_help.py:632]: up to you, easy change

Comment (mabenedi) [reply src/azure-cli/azure/cli/command_modules/acr/_help.py]: done

File: src/azure-cli/azure/cli/command_modules/acr/_params.py [hunk 3]
```diff
@@ -227,6 +227,21 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
         c.argument('force', options_list=['--force', '-f'], help='Overwrite the existing tag.', action='store_true')
         c.argument('manifest_spec', options_list=['--name', '-n'], help="The name of the artifact. May include a tag in the format 'name:tag'.")
 
+    with self.argument_context('acr cache') as c:
+        c.argument('registry_name', options_list=['--registry', '-r'])
+        c.argument('name', options_list=['--name', '-n'], help='The name of the cache rule.')
+        c.argument('cred_set', options_list=['--cred-set', '-c'], help='The name of the credential set.')
+        c.argument('source_repo', options_list=['--source-repo', '-s'], help="The full source repository path such as 'docker.io/library/ubuntu'.")
+        c.argument('target_repo', options_list=['--target-repo', '-t'], help="The target repository namespace such as 'ubuntu'.")
+        c.argument('remove_cred_set', get_three_state_flag(), help='Optional boolean indicating whether to remove the credential set from the cache rule. False by default.')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/acr/_params.py:236]: Could we use `action="store_true"` to replace `get_three_state_flag()`

**Comment by zhoxing-ms** [comment]
@mabenedi Could you please resolve these conflicts?

File: src/azure-cli/azure/cli/command_modules/acr/_params.py (hunk 3 above)
Comment (mabenedi) [reply src/azure-cli/azure/cli/command_modules/acr/_params.py]: done

File: src/azure-cli/azure/cli/command_modules/acr/commands.py [hunk 4]
```diff
@@ -224,6 +236,30 @@ def load_command_table(self, _):
         g.command('list-deleted-tags', 'acr_manifest_deleted_tags_list')
         g.command('restore', 'acr_manifest_deleted_restore')
 
+    with self.command_group('acr cache', acr_cache_util) as g:
+        g.show_command('show', 'acr_cache_show')
+        g.command('create', 'acr_cache_create')
+        g.command('list', 'acr_cache_list')
+        g.command('delete', 'acr_cache_delete')
```
[the remark this answers is not in the document]
Comment (mabenedi) [reply src/azure-cli/azure/cli/command_modules/acr/commands.py]: done

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**